### DISCUSSION TITLE
Expanding Weight Initialization Module

### DISCUSSION THREAD

**Discussion by LazaroHurtado** [LazaroHurtado/micrograd-rs#3] [Ideas]
Using the proper weight initialization can significantly boost performance gains by accelerating convergence of SGD. With this being said, we should expand our `WeightInit` enum by incorporating the He and LeCun schemes. Currently, we only offer Glorot/Xavier but this is suboptimal for the ReLu activation function so some tiny changes must be made.

Anyone can claim implementing these features:
- He Weight Initialization
- LeCun Weight Initialization

Here are some extra resources you might find helpful:
[Glorot and He discussion](https://mmuratarat.github.io/2019-02-25/xavier-glorot-he-weight-init)
[wandb Intro to Weight Inits](https://wandb.ai/sauravmaheshkar/initialization/reports/A-Gentle-Introduction-To-Weight-Initialization-for-Neural-Networks--Vmlldzo2ODExMTg)

**Comment by jrego43** [comment]
I'll try my hand at implementing He Initialization tomorrow.

**Comment by Demba3** [comment]
I will be implementing LeCun weight initialization. Please let me know if this causes any conflicts.
